src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleDocsSubmitScreen.java

```diff
@@ -156,57 +115,64 @@ private void loadDataInBackground() {
+    SwingWorker<Object, Void> worker =
         new SwingWorker<>() {
           @Override
-          protected List<Patrimoine> doInBackground() {
-            var ids = extractInputIds();
-            List<String> codePatrimoinesVisualisables = new ArrayList<>();
-            for (var id : ids) {
-              var code = googleApi.readDocsContent(authDetails, id);
-              codePatrimoinesVisualisables.add(code);
-            }
-            List<Patrimoine> patrimoinesVisualisables = new ArrayList<>();
-            PatrimoineCompiler patrimoineCompiler = new PatrimoineCompiler();
-            for (String codePatrimoine : codePatrimoinesVisualisables) {
-              String className = new ClassNameExtractor().apply(codePatrimoine);
-
-              Patrimoine patrimoineVisualisable =
-                  patrimoineCompiler.apply(className, codePatrimoine);
-
-              patrimoinesVisualisables.add(patrimoineVisualisable);
-            }
-            return patrimoinesVisualisables;
+          protected List<Map<String, String>> doInBackground() {
+            return extractInputData();
           }
 
           @Override
           protected void done() {
             loadingDialog.dispose();
             try {
-              final List<Patrimoine> patrimoinesVisualisables = get();
-              openResultFrame(patrimoinesVisualisables);
+              final List<Map<String, String>> docsLink = (List<Map<String, String>>) get();
+              openResultFrame(docsLink);
             } catch (InterruptedException | ExecutionException e) {
               throw new RuntimeException(e);
             }
           }
         };
 
-    // Show the loading dialog and start the worker
     worker.execute();
-    loadingDialog.setVisible(true); // This will block until loadingDialog is disposed
+    loadingDialog.setVisible(true);
   }
 
-  private List<String> extractInputIds() {
-    List<String> ids = new ArrayList<>();
-    for (JTextField field : inputFields) {
-      String rawLink = field.getText();
-      var parsedId = linkIdParser.apply(rawLink);
-      ids.add(parsedId);
+  private List<Map<String, String>> extractInputData() {
+    List<Map<String, String>> linkDataList = new ArrayList<>();
+
+    for (JTextArea field : inputFields) {
+      String rawText = field.getText();
+
+      String[] lines = rawText.split("\n");
+
+      for (String line : lines) {
+        String[] parts = line.split(":", 2);
+
+        if (parts.length == 2) {
+          String linkName = parts[0].trim();
+          String linkValue = parts[1].trim();
+
+          Map<String, String> linkData = new HashMap<>();
+          linkData.put("name", linkName);
+          linkData.put("link", linkValue);
+
+          linkDataList.add(linkData);
+        }
+      }
     }
-    return ids;
+
+    return linkDataList;
+  }
+
+  @SneakyThrows
+  private GoogleApi.GoogleAuthenticationDetails handleGoogleSignIn() {
+    return googleApi.requestAuthentication();
```

Comment:
i do not know  why you moved this here, could you explain?

Reply:
Sure, now it's in the GoogleDocsLinkVerfierScreen screen that we need the authentication details, and so in we need the handleGoogleSignIn method in the GoogleDocsSubmitScreen screen because that's where we're going to pass the authentication details to the GoogleDocsLinkVerfierScreen screen